src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/DefaultCache.java

```diff
@@ -233,6 +249,64 @@ public void send304(
         counters.responses.getAsLong();
     }
 
+    public boolean isRequestCacheable(
+        ListFW<HttpHeaderFW> headers)
+    {
+        return canRequestBeStored() &&
+               !headers.anyMatch(h ->
+                                 {
+                                     final String name = h.name().asString();
+                                     final String value = h.value().asString();
+                                     switch (name)
+                                     {
+                                         case CACHE_CONTROL:
+                                             return value.contains(CacheDirectives.NO_STORE);
+                                         case METHOD:
+                                             return !HttpMethods.GET.equalsIgnoreCase(value);
+                                         case TRANSFER_ENCODING:
+                                             return true;
+                                         default:
+                                             return false;
+                                     }
+                                 });
+    }
+
+    public void purgeEntriesForNonPendingRequests()
+    {
+        cachedEntries.forEach((requestHash, cacheEntry) ->
+        {
+             if (cacheEntry.getSubscribers() == 0)
+             {
+                 this.purge(requestHash);
+             }
+        });
+    }
+
+    private boolean canRequestBeStored()
+    {
+        int availableSlot = totalSlots - cacheBufferPool.acquiredSlots();
+        int availableCacheCapacityInPercentage = (availableSlot * 100) / totalSlots;
+        return allowedCachePercentage <= availableCacheCapacityInPercentage;
```

Comment:
Instead of calculating the percentage and comparing with the limit, pre-calculate the number of slots corresponding to the the maximum percentage, and ensure that acquireSlots is less than or equal to that pre-computed value.

Reply:
Thanks! I like it :)